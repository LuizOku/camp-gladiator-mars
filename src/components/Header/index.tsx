import React from "react";

import { Link } from "react-router-dom";

import { Container, Logo } from "./styles.css";

const Header = ({ testId }: { testId?: string }) => {
  return (
    <Container data-testid={testId}>
      <Link to="/">
        <Logo
          data-testid="header-logo-img"
          src={require("../../assets/img/camp-gladiator-full.png")}
        />
      </Link>
    </Container>
  );
};

export default Header;
