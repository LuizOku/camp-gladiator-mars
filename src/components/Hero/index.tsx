import React from "react";

import { Container, Text } from "./styles.css";

const Hero = ({ testId }: { testId?: string }) => {
  return (
    <Container data-testid={testId}>
      <Text type="h3" variation="display" font="united">
        Camp Gladiator is going to Mars...
      </Text>
      <Text type="h3" variation="default" font="united">
        Start now your virtual workouts!
      </Text>
    </Container>
  );
};

export default Hero;
