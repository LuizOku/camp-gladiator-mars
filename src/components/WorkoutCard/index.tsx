import React from "react";

import { Paragraph } from "@campgladiator/cgui-core.atoms.typography";
import { useNavigate } from "react-router-dom";

import { WorkoutI } from "../../shared/interfaces/workout.interface";
import {
  Container,
  ContentInfo,
  StyledAvatar,
  TagsContainer,
  StyledTag,
} from "./styles.css";

export type WorkoutCardI = {
  workout: WorkoutI;
  testId?: string;
};
const WorkoutCard = ({ workout, testId }: WorkoutCardI) => {
  const navigate = useNavigate();
  const navigateToWorkoutDetail = () => navigate(`workouts/${workout.id}`);
  return (
    <Container data-testid={testId} raised onClick={navigateToWorkoutDetail}>
      <StyledAvatar
        src={workout.thumbnail}
        size="50px"
        data-testid={`thumbnail-${workout.title}`}
        alt={`thumbnail-${workout.title}`}
      />
      <ContentInfo>
        <Paragraph size="default" weight="bold">
          {workout.title}
        </Paragraph>
        <TagsContainer>
          <StyledTag size="tiny" emphasis="primary" variation="alternative">
            Level {workout.levelTag}
          </StyledTag>
          <StyledTag size="tiny" emphasis="default">
            Impact {workout.impactTag}
          </StyledTag>
          <StyledTag size="tiny" emphasis="default" variation="alternative">
            {workout.duration} mins
          </StyledTag>
        </TagsContainer>
      </ContentInfo>
    </Container>
  );
};

export default WorkoutCard;
