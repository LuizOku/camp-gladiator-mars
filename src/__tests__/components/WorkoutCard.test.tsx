import { render, screen } from "@testing-library/react";
import { MemoryRouter as Router } from "react-router-dom";

import { WorkoutCard } from "../../components";

import "../../setupTests";
import {
  ImpactTagT,
  LevelTagT,
} from "../../shared/interfaces/workout.interface";

const baseProps = {
  workout: {
    description:
      "Enjoy a fun and high energy workout with Amber. We will engage a full body workout and get your heart rate up!",
    thumbnail:
      "https://cgcdn.s3.amazonaws.com/nation/users/UID-W1405610919UID-53c7eba7a992b.png",
    levelTag: "beginner" as LevelTagT,
    media: "https://virtual-library.s3.amazonaws.com/Week2-AmberAutrey.mp4",
    title: "Strength and Agility by Amber Autrey",
    impactTag: "high" as ImpactTagT,
    id: "3",
    trainerId: "3",
    duration: 60,
  },
};

const renderComponent = (customProps?: { testId?: string }) => {
  const props = { ...baseProps, ...customProps };

  return render(
    <Router>
      <WorkoutCard {...props} />
    </Router>
  );
};

describe("<WorkoutCard />", () => {
  it("renders without crashing", () => {
    const view = renderComponent();
    expect(view).toMatchSnapshot();
  });

  it("should have the avatar with the thumbnail", () => {
    renderComponent();
    expect(
      screen.getByTestId(`thumbnail-${baseProps.workout.title}`)
    ).toBeInTheDocument();
  });

  it("should have the workout title", () => {
    renderComponent();
    expect(screen.getByText(baseProps.workout.title)).toBeInTheDocument();
  });

  it("should have the workout level tag", () => {
    renderComponent();
    expect(
      screen.getByText(`Level ${baseProps.workout.levelTag}`)
    ).toBeInTheDocument();
  });

  it("should have the workout impact tag", () => {
    renderComponent();
    expect(
      screen.getByText(`Impact ${baseProps.workout.impactTag}`)
    ).toBeInTheDocument();
  });

  it("should have the workout duration tag", () => {
    renderComponent();
    expect(
      screen.getByText(`${baseProps.workout.duration} mins`)
    ).toBeInTheDocument();
  });

  it("should have a data-test-id", () => {
    const testId = "test-workout-card";
    renderComponent({ testId });
    const element = screen.getByTestId(testId);
    expect(element).toBeInTheDocument();
  });
});
