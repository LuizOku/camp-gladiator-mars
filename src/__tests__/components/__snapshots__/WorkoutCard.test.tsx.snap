// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WorkoutCard /> renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <article
        class="card false card--raised sc-eCYdqJ fmqJBb"
      >
        <img
          class="avatar sc-gKXOVf efHPNN"
          data-testid="thumbnail-Strength and Agility by Amber Autrey"
          src="https://cgcdn.s3.amazonaws.com/nation/users/UID-W1405610919UID-53c7eba7a992b.png"
          style="width: 50px; height: 50px;"
        />
        <div
          class="sc-jSMfEi hEqNgO"
        >
          <p
            class="paragraph "
            data-size="default"
            data-weight="bold"
          >
            Strength and Agility by Amber Autrey
          </p>
          <div
            class="sc-iBkjds cdystf"
          >
            <div
              class="tag sc-ftvSup itDdzq"
              data-emphasis="primary"
              data-size="tiny"
              data-variation="alternative"
            >
              <p
                class="paragraph "
                data-size="tiny"
                data-weight="bold"
              >
                Level 
                beginner
              </p>
            </div>
            <div
              class="tag sc-ftvSup itDdzq"
              data-emphasis="default"
              data-size="tiny"
              data-variation="default"
            >
              <p
                class="paragraph "
                data-size="tiny"
                data-weight="bold"
              >
                Impact 
                high
              </p>
            </div>
            <div
              class="tag sc-ftvSup itDdzq"
              data-emphasis="default"
              data-size="tiny"
              data-variation="alternative"
            >
              <p
                class="paragraph "
                data-size="tiny"
                data-weight="bold"
              >
                60
                 mins
              </p>
            </div>
          </div>
        </div>
      </article>
    </div>
  </body>,
  "container": <div>
    <article
      class="card false card--raised sc-eCYdqJ fmqJBb"
    >
      <img
        class="avatar sc-gKXOVf efHPNN"
        data-testid="thumbnail-Strength and Agility by Amber Autrey"
        src="https://cgcdn.s3.amazonaws.com/nation/users/UID-W1405610919UID-53c7eba7a992b.png"
        style="width: 50px; height: 50px;"
      />
      <div
        class="sc-jSMfEi hEqNgO"
      >
        <p
          class="paragraph "
          data-size="default"
          data-weight="bold"
        >
          Strength and Agility by Amber Autrey
        </p>
        <div
          class="sc-iBkjds cdystf"
        >
          <div
            class="tag sc-ftvSup itDdzq"
            data-emphasis="primary"
            data-size="tiny"
            data-variation="alternative"
          >
            <p
              class="paragraph "
              data-size="tiny"
              data-weight="bold"
            >
              Level 
              beginner
            </p>
          </div>
          <div
            class="tag sc-ftvSup itDdzq"
            data-emphasis="default"
            data-size="tiny"
            data-variation="default"
          >
            <p
              class="paragraph "
              data-size="tiny"
              data-weight="bold"
            >
              Impact 
              high
            </p>
          </div>
          <div
            class="tag sc-ftvSup itDdzq"
            data-emphasis="default"
            data-size="tiny"
            data-variation="alternative"
          >
            <p
              class="paragraph "
              data-size="tiny"
              data-weight="bold"
            >
              60
               mins
            </p>
          </div>
        </div>
      </div>
    </article>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
